pkg/apis/admissionregistration/types.go

```diff
@@ -123,8 +124,78 @@ type ValidatingAdmissionPolicy struct {
 	metav1.ObjectMeta
 	// Specification of the desired behavior of the ValidatingAdmissionPolicy.
 	Spec ValidatingAdmissionPolicySpec
+	// The status of the ValidatingAdmissionPolicy, including warnings that are useful to determine if the policy
+	// behaves in the expected way.
+	// Populated by the system.
+	// Read-only.
+	// +optional
+	Status ValidatingAdmissionPolicyStatus
+}
+
+// ValidatingAdmissionPolicyStatus represents the status of an admission validation policy.
+type ValidatingAdmissionPolicyStatus struct {
+	// The generation observed by the controller.
+	// +optional
+	ObservedGeneration int64
+	// The results of type checking for each expression.
+	// Presence of this field indicates the completion of the type checking.
+	// +optional
+	TypeChecking *TypeChecking
+	// The conditions represent the latest available observations of a policy's current state.
+	// +optional
+	// +listType=map
+	// +listMapKey=type
+	Conditions []ValidatingAdmissionPolicyCondition
```

Comment:
prefer using metav1.Condition in new types (xref https://github.com/kubernetes/enhancements/tree/master/keps/sig-api-machinery/1623-standardize-conditions) rather than defining new essentially identical condition types:

https://github.com/kubernetes/kubernetes/blob/fcfe5dfc2177320833caa451b401f2d1dc5668e7/staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/types.go#L1466-L1481

that will also let you use the already-written metav1validation.ValidateConditions() and condition helper methods in `k8s.io/apimachinery/pkg/api/meta`

Reply:
Thank you very much for pointing out the condition type KEP. Will update the conditions according to the KEP.

Reply:
Changing to use `metav1.Condition`